api/v2/views/instance_command.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.response import Response
+from api.v2.views.base import AuthViewSet
+from rest_framework import status
+from rest_framework import serializers
+from api.v2.exceptions import failure_response
+from threepio import logger
+from core.models.instance import Instance
+from api.v2.serializers.details.instance_command import InstanceCommandSerializer, POST_InstanceCommandSerializer, COMMANDS
+
+
+class InstanceCommandViewSet(AuthViewSet):
+    """
+    The InstanceCommand viewset is part of a larger "Subspace 2.0" feature:
+
+    Todo for full feature completion:
+    - LIST: A list of commands (dynamically generated?) available for an instance.
+    - CREATE: User is issuing a command from the list of command, validate accordingly and create necessary celery tasks/event hooks/etc. to start the process in motion.
+    """
+    serializer_class = InstanceCommandSerializer
+
+    def list(self, request):
+        serializer = InstanceCommandSerializer(COMMANDS, many=True)
+        return Response(serializer.data)
+
+    def create(self, request):
+        request_data = request.data
+        serializer = POST_InstanceCommandSerializer(
+            data=request_data, context={'request': request})
+        if not serializer.is_valid():
+            return Response(
+                serializer.errors,
+                status=status.HTTP_400_BAD_REQUEST)
+        try:
+            data = serializer.save()
+            if data['status'] == 'success':
+                data["message"] = "Successfully executed command."
+                return Response(data, status=status.HTTP_201_CREATED)
+            else:
+                data["message"] = "Failed to execute command."
+                return failure_response(
+                    status.HTTP_409_CONFLICT, data)
+        except Exception as exc:
+            logger.exception(exc)
+            return failure_response(
+                status.HTTP_409_CONFLICT, exc.message)
+        request_user = request.user
```

Comment:
Everything after this is unreachable.

Reply:
Code coverage for this method:

https://sonarcloud.io/component?id=atmosphere-master%3Aservice%2Ftasks%2Fdriver.py&line=1523

Reply:
Looks like no code coverage for this method yet so that's we haven't caught it.

https://sonarcloud.io/component_measures?id=atmosphere-master&metric=uncovered_lines&selected=atmosphere-master%3Aapi%2Fv2%2Fviews%2Finstance_command.py&view=tree